### PR TITLE
Add asyncinit

### PR DESCRIPTION
**Pull request by vladsaveliev**
Adds asyncinit recipe, generated from the pypi package https://pypi.org/project/asyncinit/

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/asyncinit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/kchmck/pyasyncinit
+  summary: Class decorator to enable async __init__
+  doc_url: https://pythonhosted.org/asyncinit/
+  license: MIT
+  license_file:
```
Comment (BastianZim) [inline recipes/asyncinit/meta.yaml:38]: This should probably be changed to `license_file: License` and then you'll need to download the file from the repo (https://github.com/kchmck/pyasyncinit/blob/master/License) and include it the package folder.

Comment (BastianZim) [reply recipes/asyncinit/meta.yaml:38]: Let me know if you need any help with that! :)

Comment (vladsaveliev) [reply recipes/asyncinit/meta.yaml:38]: Ah, didn't realize I need to download it - thought it would find it! Thanks so much for the tip, I'll add that :)

Comment (BastianZim) [reply recipes/asyncinit/meta.yaml]: Pleasure! Yeah it depends on whether it's included in the original package or not. (https://conda-forge.org/docs/maintainer/adding_pkgs.html#packaging-the-license-manually)

**Review by dopplershift — APPROVED** [review]

**Pull request merged**